### ISSUE TITLE
Scripts generating WPF fail after upgrading to PowerShell 7.3.0-rc.1

### ISSUE THREAD

**Issue by wi-fr** [PowerShell/PowerShell#18414]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Scripts generating WPF that had work on all PowerShell versions, including the PowerShell 7.3.0 Previews, fail after upgrading to PowerShell 7.3.0-rc.1.

### Expected behavior

```console
PS> [System.Windows.Window]::new().ShowDialog()

WPF Window pops up
```

### Actual behavior

```console
PS> [System.Windows.Window]::new().ShowDialog()

MethodInvocationException: Exception calling "ShowDialog" with "0" argument(s): "Unable to load DLL 'PenIMC_cor3.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)"
```

### Error details

```console
PS> Get-Error

Exception             : 
    Type           : System.Management.Automation.MethodInvocationException
    ErrorRecord    : 
        Exception             : 
            Type    : System.Management.Automation.ParentContainsErrorRecordException
            Message : Exception calling "ShowDialog" with "0" argument(s): "Unable to load DLL      
'PenIMC_cor3.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)"
            HResult : -2146233087
        CategoryInfo          : NotSpecified: (:) [], ParentContainsErrorRecordException
        FullyQualifiedErrorId : DllNotFoundException
        InvocationInfo        : 
            ScriptLineNumber : 1
            OffsetInLine     : 2
            HistoryId        : -1
            Line             :  [System.Windows.Window]::new().ShowDialog()
            PositionMessage  : At line:1 char:2
                               +  [System.Windows.Window]::new().ShowDialog()
                               +  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            CommandOrigin    : Internal
        ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
    TargetSite     : 
        Name          : ConvertToMethodInvocationException
        DeclaringType : System.Management.Automation.ExceptionHandlingOps, 
System.Management.Automation, Version=7.3.0.101, Culture=neutral, PublicKeyToken=31bf3856ad364e35   
        MemberType    : Method
        Module        : System.Management.Automation.dll
    Message        : Exception calling "ShowDialog" with "0" argument(s): "Unable to load DLL       
'PenIMC_cor3.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)"
    Data           : System.Collections.ListDictionaryInternal
    InnerException : 
        Type       : System.DllNotFoundException
        Message    : Unable to load DLL 'PenIMC_cor3.dll' or one of its dependencies: The specified 
module could not be found. (0x8007007E)
        TargetSite : 
            Name          : CreateResetEvent
            DeclaringType : MS.Win32.Penimc.UnsafeNativeMethods, PresentationCore, Version=7.0.0.0, 
Culture=neutral, PublicKeyToken=31bf3856ad364e35
            MemberType    : Method
            Module        : PresentationCore.dll
        Source     : PresentationCore
        HResult    : -2146233052
        StackTrace : 
   at MS.Win32.Penimc.UnsafeNativeMethods.CreateResetEvent(IntPtr& handle)
   at System.Windows.Input.PenThreadWorker..ctor()
   at System.Windows.Input.PenThreadPool.GetPenThreadForPenContextHelper(PenContext penContext)       
   at System.Windows.Input.StylusWisp.WispTabletDeviceCollection.UpdateTabletsImpl()
   at System.Windows.Input.StylusWisp.WispTabletDeviceCollection.UpdateTablets()
   at System.Windows.Input.StylusWisp.WispTabletDeviceCollection..ctor()
   at System.Windows.Input.StylusWisp.WispLogic.get_WispTabletDevices()
   at System.Windows.Input.StylusWisp.WispLogic.RegisterHwndForInput(InputManager inputManager,       
PresentationSource inputSource)
   at System.Windows.Interop.HwndStylusInputProvider..ctor(HwndSource source)
   at System.Windows.Interop.HwndSource.Initialize(HwndSourceParameters parameters)
   at System.Windows.Window.CreateSourceWindow(Boolean duringShow)
   at System.Windows.Window.ShowHelper(Object booleanBox)
   at System.Windows.Window.Show()
   at System.Windows.Window.ShowDialog()
   at CallSite.Target(Closure, CallSite, Object)
    Source         : System.Management.Automation
    HResult        : -2146233087
    StackTrace     : 
   at System.Management.Automation.ExceptionHandlingOps.ConvertToMethodInvocationException(Exception  
exception, Type typeToThrow, String methodName, Int32 numArgs, MemberInfo memberInfo)
   at CallSite.Target(Closure, CallSite, Object)
   at System.Dynamic.UpdateDelegates.UpdateAndExecute1[T0,TRet](CallSite site, T0 arg0)
   at System.Management.Automation.Interpreter.DynamicInstruction`2.Run(InterpretedFrame frame)       
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame   
frame)
CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
FullyQualifiedErrorId : DllNotFoundException
InvocationInfo        : 
    ScriptLineNumber : 1
    OffsetInLine     : 2
    HistoryId        : -1
    Line             :  [System.Windows.Window]::new().ShowDialog()
    PositionMessage  : At line:1 char:2
                       +  [System.Windows.Window]::new().ShowDialog()
                       +  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
```

### Environment data

```powershell
PS> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.0-rc.1
PSEdition                      Core
GitCommitId                    7.3.0-rc.1
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}       
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by ALIENQuake** [comment]
@wi-fr Hi,

I cannot reproduce the problem, PowerShell updated to 7.3.0-rc.1:
```
using namespace System.Diagnostics
using namespace System.Windows
using namespace System.Windows.Controls
using namespace System.Windows.Threading
using namespace WindowsBase

$wpfTypes = Add-Type -AssemblyName PresentationFramework -PassThru
$windowsBaseTypes = Add-Type -AssemblyName WindowsBase -PassThru 
$windowsBaseAssembly = $windowsBaseTypes[0].Assembly.Location
$wpfAssembly = $wpfTypes[0].Assembly.Location
$presentationCoreTypes = Add-Type -AssemblyName PresentationCore -PassThru
$presentationCoreAssembly = $presentationCoreTypes[0].Assembly.Location
$systemXamlTypes = Add-Type -AssemblyName System.Xaml -PassThru
$systemXamlAssembly = $systemXamlTypes[0].Assembly.Location
[System.Windows.Window]::new().ShowDialog()
```
![image](https://user-images.githubusercontent.com/365255/198865480-720edda1-55fb-41af-b90f-511d60f339f9.png)

**Comment by MartinGC94** [comment]
This reproduces on my Surface pro 8 but not my desktop PC. Judging by the dll name you probably need a pen compatible PC to trigger this error. It also happens when you use the built in GUI related commands (Show-Command, Get-Help -ShowWindow, etc.)

**Comment by SteveL-MSFT** [comment]
@adityapatwardhan we should follow-up with the .NET team on how to resolve this issue

**Comment by adityapatwardhan** [comment]
@wi-fr @MartinGC94 - which package did you install for RC1? MSI / Zip? If zip which one?

**Comment by daxian-dbw** [comment]
I'm able to reproduce this on my Surface 7 pro. I think this will reproduce on all Windows devices that support touch screen.
My installation was using x64 MSI package.

**Comment by MartinGC94** [comment]
I'm also using the x64 MSI package.

**Comment by ALIENQuake** [comment]
I also was able to reproduce this on my Surface Pro using the x64 MSI package. It works using 'PowerShell-7.3.0-rc.1-win-fxdependentWinDesktop' package with the installed required NET 7 RC2.

**Comment by wi-fr** [comment]
@adityapatwardhan, it is at least an issue with the store version of RC1: 
`C:\Program Files\WindowsApps\Microsoft.PowerShellPreview_7.3.101.0_x64__8wekyb3d8bbwe  `

**Comment by iSazonov** [comment]
Fixed in #18438

**Comment by daxian-dbw** [comment]
Yes, I have verified the private bits produced by the CI of https://github.com/PowerShell/PowerShell/pull/18438 on my Surface Pro 7.
